### DISCUSSION TITLE
HTML form vs AspNetCore EditForm

### DISCUSSION THREAD

**Discussion by mdmontesinos** [oqtane/oqtane.framework#4222] [General] [answered]
I was wondering if there's any specific reason why Oqtane uses a native html form with custom js validation via interop instead of an EditForm. Specially for the edit component of the module template, where you actually have a model to bind to and could add data annotations or fluent validations. Is there any drawback that I'm not aware of?

**Comment by sbwalker** [comment, marked as answer]
@mdmontesinos Oqtane was created before the EditForm was introduced in Blazor. Also I prefer to use primitive web elements in UIs rather than abstractions, as it is difficult to diagnose issues when you don't understand what is happening under the covers (ie. "black box" behavior). UI components like EditForm also tend to provide less flexibility in UI layout or styling. 

That being said, developers have the flexibility to develop components based on their individual preference in Oqtane... so you can definitely use EditForm if you want to in your modules. Or you can use a third party UI component library.  There is no restriction.

**Reply by mdmontesinos** [reply]
Thanks for the answer, it was mainly out of curiosity and just in case there was any major drawback that I wasn't aware of. I'm actually migrating the forms to EditForm in my modules and templates and everything is going fine.
